pkg/flowaggregator/flowaggregator_test.go

```diff
@@ -278,3 +282,53 @@ func createElement(name string, enterpriseID uint32) ipfixentities.InfoElementWi
 	ieWithValue, _ := ipfixentities.DecodeAndCreateInfoElementWithValue(element, nil)
 	return ieWithValue
 }
+
+func TestFlowAggregator_updateFlowAggregator(t *testing.T) {
+
+	dbEP := &clickhouseclient.ClickHouseExportProcess{}
+	opt := &util.Options{}
+	testcases := []struct {
+		FlowCollectorEnable       bool
+		FlowCollectorAddress      string
+		ClickHouseEnable          bool
+		externalFlowCollectorAddr string
+		message                   []updateParam
+		dbExportProcess           *clickhouseclient.ClickHouseExportProcess
+	}{
+		{true, "10.10.10.10:155", false, "", []updateParam{updateExternalFlowCollectorAddr}, nil},
+		{false, "", true, "addr", []updateParam{disableFlowCollector}, nil},
+		{false, "", true, "", []updateParam{enableClickHouse}, nil},
+		{false, "", true, "", []updateParam{updateClickHouseParam}, dbEP},
+		{true, "10.10.10.10:155", false, "", []updateParam{updateExternalFlowCollectorAddr, disableClickHouse}, dbEP},
+	}
+
+	timeout := time.After(1 * time.Minute)
+	done := make(chan bool)
+	go func() {
+		for i, tc := range testcases {
+			t.Run(fmt.Sprintf("subtest: %d", i), func(t *testing.T) {
+				FlowAggregator := &flowAggregator{
+					updateCh: make(chan updateMsg, 2),
+				}
+				var flowAggregatorConf flowaggregatorconfig.FlowAggregatorConfig
+				flowAggregatorConf.FlowCollector.Enable = tc.FlowCollectorEnable
+				flowAggregatorConf.FlowCollector.Address = tc.FlowCollectorAddress
+				flowAggregatorConf.ClickHouse.Enable = tc.ClickHouseEnable
+				FlowAggregator.externalFlowCollectorAddr = tc.externalFlowCollectorAddr
+				FlowAggregator.dbExportProcess = tc.dbExportProcess
+				go FlowAggregator.updateFlowAggregator(&flowAggregatorConf, opt)
+				for _, message := range tc.message {
+					msg := <-FlowAggregator.updateCh
+					assert.Equal(t, message, msg.param)
+				}
+			})
+		}
+		done <- true
+	}()
+
+	select {
+	case <-timeout:
+		t.Fatal("Test didn't finish in time")
+	case <-done:
+	}
```

Comment:
I am not sure why we are using a timeout here. Even though `updateFlowAggregator` writes to a channel, we can just use a large buffer and it will never block. Plus any timeout that large for a unit test is a red flag.

probably a better way to write a test like this:

```go
t.Run(fmt.Sprintf("subtest: %d", i), func(t *testing.T) {
    flowAggregator := &flowAggregator{
        // use a buffer large enough so that updateFlowAggregator can never block
        updateCh: make(chan updateMsg, 100),
    }
    flowAggregatorConf := flowaggregatorconfig.FlowAggregatorConfig{
        // ...
    }
    flowAggregator.updateFlowAggregator(&flowAggregatorConf, opt)
    msgs := make([]updateParam)
    for msg := range flowAggregator.updateCh{
        msgs = append(msgs, msg)
    }
    assert.ElementsMatch(t, tc.messages, msgs)
}
```

Reply:
I was using timeout for other test cases (removed) to prevent it from blocking. I've remove it.
For the "range flowAggregator.updateCh", if the channel is not closed, the msg will be blocked. So I still use "range tc.message" to compare the expected/actual value in the channel.